community/bolt/src/docs/dev/messaging.asciidoc

```diff
@@ -114,8 +114,15 @@ B1 01 8C 4D  79 43 6C 69  65 6E 74 2F  31 2E 30 A3
 ==== RESET
 
 The `RESET` message is a client message used to return the current session to a "clean" state.
+It will cause the session to `IGNORE` any message it is currently processing, as well as any message before `RESET` that had not yet begun processing.
```

Comment:
That makes sense. So it's kind of like a forced failure from the user.

But as this isn't an actual failure, it still feels to me like `FAILURE` is probably not quite the right response for the statement that is terminated and we need something new. It would likely be helpful to distinguish "failed due to an error condition" and "stopped due to client request". That would change your second example to something like:

```
RUN "stmt 2"  > TERMINATED "Action terminated due to client request for reason %s"
PULL_ALL      > IGNORED
RUN "stmt 1"  > IGNORED
PULL_ALL      > IGNORED
RESET         > SUCCESS
```